packages/services/src/view-service/fees.ts

```diff
@@ -0,0 +1,40 @@
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import {
+  TransactionPlannerRequest,
+  TransactionPlannerRequest_Output,
+  TransactionPlannerRequest_Swap,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+
+export const extractAltFee = (request: TransactionPlannerRequest): AssetId | undefined => {
+  const fields = [
+    { name: 'outputs', value: request.outputs },
+    { name: 'swaps', value: request.swaps },
+    { name: 'swapClaims', value: request.swapClaims },
+    { name: 'delegations', value: request.delegations },
+    { name: 'undelegations', value: request.undelegations },
+    { name: 'undelegationClaims', value: request.undelegationClaims },
+    { name: 'ibcRelayActions', value: request.ibcRelayActions },
+    { name: 'ics20Withdrawals', value: request.ics20Withdrawals },
+    { name: 'positionOpens', value: request.positionOpens },
+    { name: 'positionCloses', value: request.positionCloses },
+    { name: 'positionWithdraws', value: request.positionWithdraws },
+    { name: 'dutchAuctionScheduleActions', value: request.dutchAuctionScheduleActions },
+  ];
+
+  const nonEmptyField = fields.find(field => field.value.length > 0);
+
+  if (!nonEmptyField) {
+    console.warn('No non-empty field found in the request');
+    return undefined;
+  }
+
+  type PossibleTypes = TransactionPlannerRequest_Output | TransactionPlannerRequest_Swap;
```

Comment:
`PossibleTypes` needs to be expanded to support other actions, for instance auctions where a user can perform price discovery on a specific asset _without_ needing UM tokens.

Reply:
If `PossibleTypes` are defined not only for wasm planner but for the whole chain, it might be more convenient to import it from the core repo.

Reply:
`PossibleTypes` is a custom type I created to capture a subset of the _relevant_ action fields in `TransactionPlannerRequest`. The `assetId` will be extracted from that. I just need to extend this to extract the assetId from other types, for instance `TransactionPlannerRequest_ActionDutchAuctionSchedule`. For our purposes, I think this approach is sufficient.